### PR TITLE
Update libraries for SR2022

### PR DESCRIPTION
**Pull request by trickeydan**
Generated from srobo-robot2022.0.0.manifest using https://gist.github.com/trickeydan/ac1511cd7a0cbf7524d4926404ed1f0b

### CODE REVIEW

File: programming/python/libraries.md [hunk 1]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
+* [pandas 1.3.3](https://pypi.org/project/pandas)
+* pkg-resources 57.4.0
+* [pydantic 1.8.2](https://pypi.org/project/pydantic)
+* pygobject 3.40.1
+* [pyparsing 2.4.7](https://pypi.org/project/pyparsing)
+* [pyquaternion 0.9.9](https://pypi.org/project/pyquaternion)
+* [pyserial 3.5](https://pypi.org/project/pyserial)
+* [pytz 2021.1](https://pypi.org/project/pytz)
+* [pyusb 1.2.1](https://pypi.org/project/pyusb)
+* pyyaml 5.4.1
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:40]:
```suggestion
* [pyyaml 5.4.1](https://pypi.org/project/PyYAML/)
```

File: programming/python/libraries.md [hunk 2]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
+* [j5 0.13.0](https://pypi.org/project/j5)
+* [j5-zoloto 0.1.0](https://pypi.org/project/j5-zoloto)
+* [kiwisolver 1.3.2](https://pypi.org/project/kiwisolver)
+* [matplotlib 3.4.1](https://pypi.org/project/matplotlib)
+* [numpy 1.21.2](https://pypi.org/project/numpy)
+* opencv 4.5.2
+* [pandas 1.3.3](https://pypi.org/project/pandas)
+* pkg-resources 57.4.0
+* [pydantic 1.8.2](https://pypi.org/project/pydantic)
+* pygobject 3.40.1
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:34]:
```suggestion
* [pygobject 3.40.1](https://pypi.org/project/PyGObject/)
```

File: programming/python/libraries.md [hunk 3]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
- * [tables 3.0.0](https://pypi.python.org/pypi/tables)
- * [Tornado 2.2.1](http://www.tornadoweb.org/)
- * [urlgrabber 3.10.1](http://urlgrabber.baseurl.org)
- * [Werkzeug 0.9.4](http://werkzeug.pocoo.org/)
- * [ZeroMQ 13.0.2](http://zeromq.org/bindings:python)
+* [astoria 0.7.0](https://pypi.org/project/astoria)
+* [cached-property 1.5.2](https://pypi.org/project/cached-property)
+* [click 8.0.1](https://pypi.org/project/click)
+* [cycler 0.10.0](https://pypi.org/project/cycler)
+* dateutil 2.8.2
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:18]:
```suggestion
* [dateutil 2.8.2](https://pypi.org/project/python-dateutil/)
```

File: programming/python/libraries.md [hunk 4]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
+* [docopt 0.6.2](https://pypi.org/project/docopt)
+* [gmqtt 0.6.10](https://pypi.org/project/gmqtt)
+* [gpg 1.16.0](https://pypi.org/project/gpg)
+* [iniparse 0.5](https://pypi.org/project/iniparse)
+* [j5 0.13.0](https://pypi.org/project/j5)
+* [j5-zoloto 0.1.0](https://pypi.org/project/j5-zoloto)
+* [kiwisolver 1.3.2](https://pypi.org/project/kiwisolver)
+* [matplotlib 3.4.1](https://pypi.org/project/matplotlib)
+* [numpy 1.21.2](https://pypi.org/project/numpy)
+* opencv 4.5.2
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:30]:
```suggestion
* [opencv 4.5.2](https://pypi.org/project/opencv-python-headless/)
```
Presumably?

File: programming/python/libraries.md [hunk 5]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
+* [pyparsing 2.4.7](https://pypi.org/project/pyparsing)
+* [pyquaternion 0.9.9](https://pypi.org/project/pyquaternion)
+* [pyserial 3.5](https://pypi.org/project/pyserial)
+* [pytz 2021.1](https://pypi.org/project/pytz)
+* [pyusb 1.2.1](https://pypi.org/project/pyusb)
+* pyyaml 5.4.1
+* [rpm 4.16.1.3](https://pypi.org/project/rpm)
+* [setuptools 57.4.0](https://pypi.org/project/setuptools)
+* [six 1.16.0](https://pypi.org/project/six)
+* sr-robot3 2022.0.1
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:44]:
```suggestion
* [sr-robot3 2022.0.1](https://pypi.org/project/sr.robot3/)
```

File: programming/python/libraries.md [hunk 6]
```diff
@@ -11,69 +11,41 @@ Robot Kit
...
+* [toml 0.10.2](https://pypi.org/project/toml)
+* [typing-extensions 3.10.0.0](https://pypi.org/project/typing-extensions)
+* [udiskie 2.3.3](https://pypi.org/project/udiskie)
+* udiskie-locale-en-us 2.3.3
```
Comment (PeterJCLaw) [inline programming/python/libraries.md:48]:
```suggestion
* [udiskie-locale-en-us 2.3.3](https://pypi.org/project/udiskie/)
```
Maybe?

**Review by PeterJCLaw — APPROVED** [review]
Nice, great to have this (mostly) scripted too.
Might be worth seeing if the script can handle 302s which do renaming for case/punctuation?

`dateutil` is the only one which will be complicated of course.

I also can't find `dateutil-zoneinfo`. Any idea where that one comes from?

File: programming/python/libraries.md (hunk 4 above)
Comment (trickeydan) [reply programming/python/libraries.md:30]: We're actually compiling OpenCV ourselves, so are not using the pre-built binary you've suggested. Not actually sure what to do with this one.

Comment (PeterJCLaw) [reply programming/python/libraries.md:30]: Ah, is there a Python API available? If not, maybe just link out to OpenCV's main homepage?

Comment (trickeydan) [reply programming/python/libraries.md:30]: So the python bindings for OpenCV are generated using SWIG, so there isn't actually source to link to there.

I'm keen to keep this automated if possible, so might add a section with some hard-coded sites to that script.

**Review by sedders123 — APPROVED** [review]

File: programming/python/libraries.md (hunk 5 above)
Comment (PeterJCLaw) [reply programming/python/libraries.md]: https://github.com/srobo/docs/pull/271/commits/b5b4c9d789bd67dbfbd8be6c527bdd0b7ff3c8be

File: programming/python/libraries.md (hunk 3 above)
Comment (PeterJCLaw) [reply programming/python/libraries.md]: https://github.com/srobo/docs/pull/271/commits/b5b4c9d789bd67dbfbd8be6c527bdd0b7ff3c8be

File: programming/python/libraries.md (hunk 2 above)
Comment (PeterJCLaw) [reply programming/python/libraries.md]: https://github.com/srobo/docs/pull/271/commits/b5b4c9d789bd67dbfbd8be6c527bdd0b7ff3c8be

File: programming/python/libraries.md (hunk 1 above)
Comment (PeterJCLaw) [reply programming/python/libraries.md]: https://github.com/srobo/docs/pull/271/commits/b5b4c9d789bd67dbfbd8be6c527bdd0b7ff3c8be

**Comment by trickeydan** [comment]
(Meta: I was working on this last night and was quite surprised to find that it had been taken over and merged...)

**Pull request merged**